fuel-tx/src/transaction/consensus_parameters/gas.rs

```diff
@@ -334,6 +337,14 @@ pub struct DependentCost {
     pub dep_per_unit: Word,
 }
 
+impl DependentCost {
+    pub fn resolve(&self, units: Word) -> Word {
+        // Apply the linear transformation from units to cost:
+        // f(x) = base + mx
+        self.base + self.dep_per_unit.saturating_mul(units)
```

Comment:
The formula is `base + units/dep_pet_unit`.


```suggestion
        self.base + units.saturating_div(self.dep_per_unit)
```

<img width="703" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/e2a7479b-5120-4e65-9bc8-cbbf1c0f130a">

Reply:
How do we handle the case where `dep_per_unit` is 0, as with `DependentCost::unit()`?

Reply:
In production, we don't have cases like this=) But in the tests it overflows